File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
      Arrays.sort(intervals,(a,b)->a[0]-b[0]);
      List<int[]>res=new ArrayList<>();
    int[] current=intervals[0];
    for(int i=1;i<intervals.length;i++){
        if(intervals[i][0]<=current[1]){
            current[1]=Math.max(current[1],intervals[i][1]);
        }else{
            res.add(current);
            current=intervals[i];
        }
    }
        res.add(current);
    
    return res.toArray(new int[res.size()][]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        ans=[]

        ans.append(intervals[0])
        for start, end in intervals[1:]:
            last_end=ans[-1][1]
            if start<=last_end:
                ans[-1][1]=max(last_end,end)
            else:
                ans.append([start,end])
        return ans
